Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -524,60 +532,68 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         // TODO: we could wrap the failure here with diagnostics if it wasn't optional throughout
         defer { try? cache?.close() }
 
-        let key = try CacheKey(
-            packageIdentity: packageIdentity,
-            manifestPath: path,
-            toolsVersion: toolsVersion,
-            env: ProcessEnv.vars,
-            swiftpmVersion: SwiftVersion.currentVersion.displayString,
-            fileSystem: fileSystem
-        )
+        let key : CacheKey
+        do {
+            key = try CacheKey(
+                packageIdentity: packageIdentity,
+                manifestPath: path,
+                toolsVersion: toolsVersion,
+                env: ProcessEnv.vars,
+                swiftpmVersion: SwiftVersion.currentVersion.displayString,
+                fileSystem: fileSystem
+            )
+        } catch {
+            return completion(.failure(error))
+        }
```

Comment:
nit: this can probably move into the next do block to avoid having two?

Reply:
The second one includes this diagnostic:

```swift
            observabilityScope.emit(warning: "failed loading cached manifest for '\(key.packageIdentity)': \(error)")
```

So I thought it best to keep them separate.